lib/src/tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import styled from "styled-components";
+import TooltipPropsType from "./types";
+import * as Tooltip from "@radix-ui/react-tooltip";
+import CoreTokens from "../common/coreTokens";
+
+const DxcTooltip = ({ position = "bottom", label, children }: TooltipPropsType): JSX.Element => {
+  const SIZE = 8;
+
+  return label ? (
+    <Tooltip.Provider delayDuration={300}>
+      <Tooltip.Root>
+        <Tooltip.Trigger asChild>
+          <TooltipTriggerContainer>{children}</TooltipTriggerContainer>
+        </Tooltip.Trigger>
+        <Tooltip.Portal>
+          <StyledTooltipContent side={position} sideOffset={SIZE} data-testid={"popover-content"}>
+            <TooltipContainer>{label}</TooltipContainer>
+            <TooltipArrow asChild aria-hidden>
+              <Triangle size={SIZE} />
+            </TooltipArrow>
+          </StyledTooltipContent>
+        </Tooltip.Portal>
+      </Tooltip.Root>
+    </Tooltip.Provider>
+  ) : (
+    <>{children}</>
+  );
+};
+const Triangle = styled.span<{ size: number }>`
+  display: block;
+  height: ${(props) => `${props.size}px`};
+  width: ${(props) => `${props.size}px`};
+  background-color: ${CoreTokens.color_grey_800};
+  position: absolute;
+  left: ${(props) => `calc(50% - ${props.size / 2}px)`};
+  bottom: ${(props) =>
+    `-${props.size / 2 - 1}px`}; // Moved up 1 pixel to prevent sub-pixel alignment causing visual inconsistencies
+  clip-path: polygon(0% 0%, 100% 100%, 0% 100%);
+  transform: rotate(-45deg);
+  border-radius: 0 0 0 1px;
+`;
+
+const TooltipTriggerContainer = styled.div`
+  display: inline-flex;
+  position: relative;
+`;
+
+const StyledTooltipContent = styled(Tooltip.Content)`
+  z-index: 2147483647;
+
+  animation-duration: 0.2s;
+  animation-timing-function: ease-out;
```

Comment:
Was the animation and delay discussed?

Reply:
The animation was approved by Lara.

For the delay, I read some recommendations from multiple authors, the correct delay for a tooltip is 0.3s-0.5s (while many popular libraries set 0.5s by default) so I decided that a base delay of 0.3s and an animation of 0.2s would be appropiate. We can open a discussion regarding this topic if needed.